src/components/dialogs/SearchDialog.vue

```diff
@@ -0,0 +1,264 @@
+  formatSize!: (_: any) => string;
+  translate!: (_: any, __?: any) => string;
+
+  searchForm: {
+    valid: boolean;
+    category: string | null;
+    pattern: string | null;
+    plugin: string | null;
+  } = {
+    valid: false,
+    category: null,
+    pattern: null,
+    plugin: null,
+  };
+
+  grid: GridConfig = {
+    searchItems: [],
+    downloadItem: {
+      descrLink: "",
+      fileName: "",
+      fileSize: 0,
+      fileUrl: "",
+      nbLeechers: 0,
+      nbSeeders: 0,
+      siteUrl: "",
+    },
+    headers: [
+      { text: this.translate("name"), value: "fileName" },
+      { text: this.translate("size"), value: "fileSize" },
+      { text: this.translate("seeds"), value: "nbSeeders" },
+      { text: this.translate("peers"), value: "nbLeechers" },
+      { text: this.translate("search_engine"), value: "siteUrl" },
+      { text: this.translate("action", 2), value: "actions", sortable: false },
+    ],
+  };
+
+  loading = false;
+
+  setPasteUrl!: (_: any) => void;
+  openAddForm!: () => void;
+  addFormDownloadItem!: (_: any) => void;
+
+  async mounted() {
+    this.availablePlugins = await this.getAvailablePlugins();
+  }
+
+  async downloadTorrent(item: SearchTaskTorrent) {
+    this.addFormDownloadItem({
+      downloadItem: {
+        title: item.fileName,
+        url: item.fileUrl,
+      },
+    });
+    this.openAddForm();
+  }
+
+  async getAvailablePlugins(): Promise<SearchPlugin[]> {
+    const availablePlugins = await api.getSearchPlugins();
+
+    return availablePlugins.filter((plugin) => plugin.enabled === true);
+  }
+
+  async triggerSearch() {
+    try {
+      this.grid.searchItems = []; // Clear the table
+      this.loading = true;
+
+      const response = await this._startSearch();
+      this._searchId = response.id;
+
+      this.setTaskAndRun(this.task(response));
+    } catch {
+      //
+    }
+  }
+
+  async stopSearch() {
+    this.cancelTask();
+    this._stopSearch(this._searchId);
+    this.loading = false;
+  }
+
+  async getResults(id: number): Promise<SearchTaskTorrent[]> {
+    const response = await api.getSearchResults(id);
+    return response.results;
+  }
+
+  @Emit("input")
+  closeDialog() {
+    return false;
+  }
+
+  private async _startSearch(): Promise<{ id: number }> {
+    const result = await api.startSearch(
+      this.searchForm.pattern,
+      this.searchForm.plugin,
+      this.searchForm.category
+    );
+
+    return result;
+  }
+
+  private async _stopSearch(id: number): Promise<AxiosResponse> {
+    return await api.stopSearch(id);
+  }
+
+  /**
+   * Does request until the plugins return data
+   */
+  private task(response: { id: number }): CallableFunction {
+    return async () => {
+      const results = await this.getResults(response.id);
+
+      this.grid.searchItems = this.grid.searchItems.concat(results);
+
+      if (results && results.length) {
+        this.stopSearch();
+        return true;
+      }
+
+      return this.task(response)();
```

Comment:
Doesn't need recursively call.
Since the task will be called repeatedly by `HasTask`, and the `response` will not change.